reflex-dom-core/src/Reflex/Dom/Prerender.hs

```diff
@@ -48,7 +50,7 @@ type PrerenderClientConstraint js t m =
   , DomRenderHook t m
   , HasDocument m
   , TriggerEvent t m
-  , Prerender js t m
```

Comment:
Is it no longer possible to have this? It seems pretty important for composability of widgets.

Reply:
Yes this avoids a type error, or an infinite loop with `UndecidableSuperClasses`. It's unfortunate because the instances always terminate, I think after one round even with `Client (Client m) ~ Client m`. Maybe we can use that constraint instead, which due to the fucked up nature of type families (delaying with the stuck terms) will not cause problems an accomplish the same thing.

Reply:
It seems to work fine if you lift it into the actual class head and not the constraint synonym.